shanalikhan/code-settings-sync
# Upload UX Improvements

Task checklist (roughly):
- [x] Add status bar message for "already uploaded" case
- [x] If local and gist is different and lastLocalDownload < lastGistUpload, display popup dialog to ask for one-time force update
    - [x] Remove "Don't show again" UX and code
    - [x] Change dialog box to "The gist settings changed since you last downloaded them.  Do you want to upload your current local settings to the gist anyway?"  Single yes/no button.
- [x] If local and gist is different and lastLocalDownload >= lastGistUpload, upload with only a status bar message
- [x] NLS for new dialogs and messages
- [x] ~~Update README doc~~ (reverted to original)
- [x] Update wiki
 
#### Short description of what this resolves:
- Improve the upload user experience
  - Detect if gist has changes that are not on the user's machine and prompt for overwrite
  - Remove popup option to turn on forceUpload and "Don't show again"

#### Changes proposed in this pull request:

- Streamline Upload UX
  - Remove "Don't show this dialog again" feature and setting
  - Change "Gist is newer than what you downloaded" dialog to a simple one-time force upload button

**Fixes**: #1035

#### How Has This Been Tested?



Since only the Upload logic was changed, it was necessary to test only upload features
1. Test the simple case of the user starting with synced settings and uploading without a change.
  - [x] Should see status bar message saying settings already uploaded
2. Test the simple case of the user starting with synced settings and uploading a change.
  - [x] No dialog popup and status bar indicates the successful upload
3. Modify the gist with another client and try to upload changes from the client under test.  Expect to see a prompt asking if the user wants a forced upload.
  - [x] Verify that selecting Yes proceeds with the upload
  - [x] Verify that selecting No or closing the dialog does nothing and show canceled status msg
4. Verify that automatic upload on setting change still works.
  - [x] Start with synced settings and make a change - should complete normally
  - [x] Push a change from another host and make a change - should get prompt to overwrite
5. Upload after reset
  - [x] Reset, repeat test 1
  - [x] Reset, repeat test 2
  - [x] Reset, repeat test 3
6. Upload with forceUpload=true, All should do an upload
  - [x] Reset, repeat test 1
  - [x] Reset, repeat test 2
  - [x] Reset, repeat test 3 (expect no dialog)
7. New Gist: Delete gist, reset settings, and perform an upload
  - [x] New gist should be created and upload completes

#### Screenshots (if appropriate):
n/a

#### Checklist:
